packages/odf/components/storage-consumers/update-storage-quota-modal.tsx

```diff
@@ -0,0 +1,195 @@
+    unit: humanizedQuota.unit,
+  } as StorageQuota;
+};
+
+const updateStorageConsumer = (
+  storageConsumer: StorageConsumerKind,
+  quotaValue: number
+): Promise<StorageConsumerKind> => {
+  const patches: Patch[] = [
+    {
+      op: 'replace',
+      path: '/spec/storageQuotaInGiB',
+      value: quotaValue,
+    },
+  ];
+  return k8sPatch({
+    model: StorageConsumerModel,
+    resource: storageConsumer,
+    data: patches,
+  });
+};
+
+type UpdateStorageQuotaModalProps = CommonModalProps<{
+  resource: StorageConsumerKind;
+}>;
+
+const UpdateStorageQuotaModal: React.FC<UpdateStorageQuotaModalProps> = (
+  props
+) => {
+  const {
+    extraProps: { resource },
+    isOpen,
+    closeModal,
+  } = props;
+  const { t } = useCustomTranslation();
+  const [inProgress, setInProgress] = React.useState(false);
+  const [finished, setFinished] = React.useState(false);
+  const [error, setError] = React.useState<string>(null);
+  const initialQuota = getStorageConsumerQuotaWithoutB(
+    resource?.spec?.storageQuotaInGiB
+  );
+  const [quota, setQuota] = React.useState(initialQuota);
+
+  const updateStorageConsumerQuota = async () => {
+    try {
+      setInProgress(true);
+      setFinished(false);
+      await updateStorageConsumer(resource, getQuotaValueInGiB(quota));
+    } catch (e) {
+      setError(e);
+    } finally {
+      setInProgress(false);
+      setFinished(true);
+    }
+  };
+
+  return (
+    <Modal isOpen={isOpen} onClose={closeModal} variant={ModalVariant.medium}>
+      <ModalTitle>{t('Edit storage quota')}</ModalTitle>
+      <ModalBody>
+        {finished && (
+          <Alert
+            variant={error ? 'danger' : 'success'}
+            isInline
+            title={
+              error
+                ? t(
+                    'Storage quota request failed. Make sure your Data foundation provider cluster has enough capacity before trying again.'
+                  )
+                : t('Storage quota update successfully')
+            }
+          ></Alert>
+        )}
+        <StorageQuotaBody
+          quota={quota}
+          setQuota={setQuota}
+          initialQuota={initialQuota}
+          capacityInfo={<AvailableCapacity />}
+        />
+      </ModalBody>
+      <ModalFooter inProgress={inProgress}>
+        <Flex direction={{ default: 'row' }}>
+          <FlexItem>
+            <Button
+              key="cancel"
+              variant="secondary"
+              onClick={closeModal}
+              data-test-id="modal-cancel-action"
+            >
+              {t('Cancel')}
+            </Button>
+          </FlexItem>
+          <FlexItem>
+            <Button
+              key="save"
+              data-test="modal-submit-action"
+              data-test-id="confirm-action"
+              variant="primary"
+              onClick={updateStorageConsumerQuota}
+              isDisabled={
+                !isValidQuota(quota, initialQuota) || inProgress || finished
+              }
```

Comment:
since we are not closing the modal after clicking `Save changes`, I am assuming we want users to increase the quota again if they want (without closing and re-opening the modal)...
```suggestion
              isDisabled={
                !isValidQuota(quota, initialQuota) || inProgress || (quota.value === initialQuota.value && quota.unit === initialQuota.unit)
              }
```

Reply:
also we need to reset `setFinished` back to `false` in this case...

Reply:
@SanjalKatiyar we are not closing so they can see the success message according to the design: in the design the button is disabled.